### PR TITLE
1MB plane build reduce

### PR DESCRIPTION
**Pull request by magicrub**
For Plane this saves 14k Flash on fmuv2

Mega squash of a bunch of # if stuff to isolate a few libraries. @peterbarker Discuss! Lets figure out what we want and such and then I'll bring them out into individual library commits

This is part of the effort in https://github.com/ArduPilot/ardupilot/issues/19068

### CODE REVIEW

**Comment by hendjoshsr71** [comment]
Output for Durandal Build Size: Rover & Tracker increased by 5kB...?
image

Just a note the difference between boards with  `HAL_MINIMIZE_FEATURES 1` && <1MB and those just <1MB is around -7kB

There are only 9 boards that define `HAL_MINIMIZE_FEATURES 1`. Are we getting closer to a point where HAL_MINIMIZE_FEATURES  should just mean anything less <=1MB for certain vehicles?

**Comment by hendjoshsr71** [comment]
Trolling through the list of every where we have `!HAL_MINIMIZE_FEATURES` I'd surmise most of those items could become `BOARD_SIZE > 1024` instead.

There are a few majors ones that should not: OSD, DLVR sensors, Telemetry protocols

**Review by tridge — CHANGES_REQUESTED** [review]
thanks Tom!
A few things to do:
 - each of the options where we now disable needs to be in the custom build server build list in Tools/scripts/build_options.py so users have a way to re-enable if needed
 - we need to understand why the build size on 2M boards is reduced. What code has been removed?

File: ArduPlane/config.h [hunk 1]
```diff
@@ -29,7 +29,7 @@
 //
 
 #ifndef ADVANCED_FAILSAFE
- # define ADVANCED_FAILSAFE ENABLED
+ # define ADVANCED_FAILSAFE (!HAL_MINIMIZE_FEATURES && BOARD_FLASH_SIZE > 1024)
```
Comment (peterbarker) [inline ArduPlane/config.h:32]: We'd like to remove `HAL_MINIMIZE_FEATURES`.... 

Like parachute, I'm not sure we should start to compile out safety-critical features.

File: libraries/AP_KDECAN/AP_KDECAN.cpp [hunk 2]
```diff
@@ -606,6 +607,7 @@ bool AP_KDECAN::pre_arm_check(char* reason, uint8_t reason_len)
 
     _enum_sem.give();
 
+#if defined(AP_MOTORS_ENABLE) && AP_MOTORS_ENABLE
```
Comment (peterbarker) [inline libraries/AP_KDECAN/AP_KDECAN.cpp:610]:
```suggestion
#if defined(AP_MOTORS_ENABLED) && AP_MOTORS_ENABLED
```

**Review by peterbarker — CHANGES_REQUESTED** [review]
Would be nice to not increase the usage of `MINIMIZE_FEATURES`.

**Comment by magicrub** [comment]
This is stale.. closing

**Pull request closed without merging**